### PR TITLE
soc: arm: add support for stm32h753xx

### PR DESCRIPTION
**Pull request by petriok**
STM32H753xx is similar to STM32H743xx except that it has crypto/hash
hardware acceleration and the memory configuration is always 2Mbytes
flash and 1Mbyte RAM.

Signed-off-by: Petri Oksanen <petri@iote.ai>

### CODE REVIEW

File: soc/arm/st_stm32/stm32h7/Kconfig.defconfig.stm32h753xx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# ST STM32H753X MCU configuration options
+
+# Copyright (c) 2020 Petri Oksanen
+# petri@iote.ai
```
Comment (erwango) [inline soc/arm/st_stm32/stm32h7/Kconfig.defconfig.stm32h753xx:4]: Please use one liner format

File: dts/arm/st/h7/stm32h753.dtsi [hunk 2]
```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2020 Petri Oksanen
+ * petri@iote.ai
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <mem.h>
+#include <st/h7/stm32h7.dtsi>
```
Comment (erwango) [inline dts/arm/st/h7/stm32h753.dtsi:9]: Ass mentionned in the commit stm32h753 are basically stm32h743 with crypto/hash.
So let's make this explicit and include stm32h743 here:
```suggestion
#include <st/h7/stm32h743.dtsi>
```
And remove the redundant definitions bellow.
Flash size should be defined in dedicated package stm32h753Xi.dtsi file.

**Review by erwango — CHANGES_REQUESTED** [review]
Thanks for this contribution, some changes requested on copyright format and dtsi files definition

File: dts/arm/st/h7/stm32h753.dtsi [hunk 3]
```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2020 Petri Oksanen
+ * petri@iote.ai
```
[the remark this answers is not in the document]
Comment (petriok) [reply dts/arm/st/h7/stm32h753.dtsi]: Noted; I looked at a bad example file

File: dts/arm/st/h7/stm32h753.dtsi (hunk 2 above)
Comment (petriok) [reply dts/arm/st/h7/stm32h753.dtsi]: Thanks, starting out with a simple commit to see the process. Added the file stm32h753Xi.dtsi.

**Review by erwango — APPROVED** [review]

**Review by ABOSTM — APPROVED** [review]
LGTM

**Pull request merged**